src/language/ast.ts

```diff
@@ -670,3 +672,14 @@ export interface InputObjectTypeExtensionNode {
   readonly directives?: ReadonlyArray<ConstDirectiveNode>;
   readonly fields?: ReadonlyArray<InputValueDefinitionNode>;
 }
+
+// Schema Coordinates
+
+export interface SchemaCoordinateNode {
+  readonly kind: 'SchemaCoordinate';
+  readonly loc?: Location;
+  readonly isDirective: boolean;
```

Comment:
@leebyron `isDirective` sounds good if we reference the directive itself e.g. `@foo` but it confusing if we reference the directive argument e.g. `@foo.arg`
Maybe we can just have two node kinds `TypeCoordinate` and `DirectiveCoordinate`?
Since directives represent another namespace anyway?

Reply:
That's a reasonable suggestion, I tried it out but found it made things a lot more complicated. Created 3 types instead of 1 to represent coordinates, complicated the predicate checker, and then backporting the change to the spec text (assuming the AST should mirror the grammar) made it a little bit more awkward to explain (sets of semantics for each sub-grammar). It wasn't awful, but it's definitely a bit of cognitive cost.

Then IMHO `DirectiveCoordinate` implies something is a directive more or less equivalently as `isDirective` - or at least not more so to offset the cognitive cost of a more complex implementation and spec text.

Reply:
@leebyron Agree, with the above arguments 👍 
But still `isDirective` doesn't look right for coordinates for directive arguments.
How about renaming it to `ofDirective` meaning it either:
- schema coordinate `of directive`
- schema coordinate `of directive` argument

Reply:
Sure, that makes some sense. Let's do that